pkg/apis/runtime/v1alpha1/function_types.go

```diff
@@ -45,15 +45,35 @@ type FunctionSpec struct {
 	Env []v1.EnvVar `json:"env,omitempty"`
 }
 
+// TemplateKind defines the type of BuildTemplate used by the build.
+type FunctionCondition string
+
+const (
+	// Indicates that function has a running condition.
+	FunctionConditionRunning FunctionCondition = "Running"
+	// Indicates that function has an error condition.
+	FunctionConditionError FunctionCondition = "Error"
+	// Indicates that function has a deploying condition.
+	FunctionConditionDeploying FunctionCondition = "Deploying"
+	// Indicates that there is a new image and function needs to be updated.
+	FunctionConditionUpdating FunctionCondition = "Updating"
+)
+
 // FunctionStatus defines the observed state of Function
 type FunctionStatus struct {
+	Condition FunctionCondition `json:"condition,omitempty"`
```

Comment:
imho a list of Conditions would be nice here, so you could tell the user on a finer level whats currently happening to the function

see basically every knative CRD e.g. https://knative.dev/docs/reference/serving-api/#github.com/knative/serving/pkg/apis/serving/v1alpha1.ServiceStatus

Reply:
It will be manage in another issue.